src/main/java/uk/gov/hmcts/darts/common/repository/EventRepository.java

```diff
@@ -82,15 +82,21 @@ Page<EventSearchResult> searchEventsFilteringOn(
         Pageable pageable);
 
     @Query(value = """
-        SELECT event_id
-        FROM  darts.event
-        WHERE is_current=true
-        AND event_id <> 0
-        AND event_id IS NOT NULL
-        GROUP BY event_id
-        HAVING count(event_id) > 1
+        SELECT distinct e2.event_id
+        from (
+          SELECT e.event_id, he.eve_id, string_agg(he.hea_id::varchar, ',' order by he.hea_id) as hearing_ids FROM darts.event e
+          LEFT JOIN darts.hearing_event_ae he
+          ON he.eve_id = e.eve_id
+          where e.is_current = true
+          AND e.event_id <> 0
+          AND e.event_id IS NOT null
+          GROUP by he.eve_id, e.event_id
+        ) e2
+        GROUP BY e2.event_id, e2.hearing_ids
+        having count(e2.event_id) > 1
+        limit = :limit
         """, nativeQuery = true)
-    List<Integer> getCurrentEventIdsToBeProcessed(Pageable pageable);
+    List<Integer> getCurrentEventIdsToBeProcessed(long limit);
```

Comment:
Currently the rest of the application uses pageable instead of limit. Is there a reason you are detracting from this pattern?

Reply:
Yeah, pageable is for when we want to grab keep our place in the search, or use multiple pages of items.

Here we just process the first X amount and we never care about what is on the other pages, so limit is more performant as we don't need to count all the items, work out the number of pages, deal with offsets, etc